File: cpp/program/Petshop.cpp
#include <iostream>
#include <string>

using namespace std;

// kelas Petshop buat nyimpen data barang yang dijual di petshop 
class Petshop {
    private:
        // atribut private buat nyimpen id, nama, kategori, dan harga barang
        string id, nama, stok, harga;

    public:
        // konstruktor default, buat bikin objek tanpa diisi dulu
        Petshop() {
            this->id = "";
            this->nama = "";
            this->stok = "";
            this->harga = "";
        }

        // konstruktor pake parameter, langsung ngisi atribut saat objek dibuat
        Petshop(string id, string nama, string stok, string harga) {
            this->id = id;
            this->nama = nama;
            this->stok = stok;
            this->harga = harga;
        }

        // Getter buat ngambil id barang
        string get_id() {
            return this->id;
        }

        // Setter buat ngubah id barang
        void set_id(string id) {
            this->id = id;
        }

        // Getter buat ngambil nama barang
        string get_nama() {
            return this->nama;
        }

        // Setter buat ngubah nama barang
        void set_nama(string nama) {
            this->nama = nama;
        }

        // Getter buat ngambil kategori barang
        string get_stok() {
            return this->stok;
        }

        // Setter buat ngubah kategori barang
        void set_stok(string stok) {
            this->stok = stok;
        }

        // Getter buat ngambil harga barang
        string get_harga() {
            return this->harga;
        }

        // Setter buat ngubah harga barang
        void set_harga(string harga) {
            this->harga = harga;
        }

        // destruktor,
        ~Petshop() {
        }
};

File: python/program/Petshop.py
class Petshop:
    def __init__(self, id="", nama="", stok="", harga=""):
        self.__id = id
        self.__nama = nama
        self.__stok = stok
        self.__harga = harga

    # Getter dan Setter untuk id
    def get_id(self):
        return self.__id

    def set_id(self, id):
        self.__id = id

    # Getter dan Setter untuk nama
    def get_nama(self):
        return self.__nama

    def set_nama(self, nama):
        self.__nama = nama

    # Getter dan Setter untuk stok
    def get_stok(self):
        return self.__stok

    def set_stok(self, stok):
        self.__stok = stok

    # Getter dan Setter untuk harga
    def get_harga(self):
        return self.__harga

    def set_harga(self, harga):
        self.__harga = harga
